core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/migrate/SchemaMigration.java

```diff
@@ -0,0 +1,645 @@
+    private Map<String, ElementFilter> entitiesPostTransformFilterMap = new HashMap<>();
+    private Map<String, ElementFilter> edgesPostTransformFilterMap = new HashMap<>();
+
+    public enum MigrationOutputType {
+        NEW, OLD
+    }
+
+    @Override
+    public void preExecute(final OperationChain<?> opChain, final Context context) {
+        if (!edges.isEmpty() || !entities.isEmpty()) {
+            List<? extends Operation> opsWithViewList = opChain.flatten()
+                    .stream()
+                    .filter(OperationView::hasView)
+                    .collect(Collectors.toList());
+
+            for (final Operation op : opsWithViewList) {
+                entitiesPostAggregationFilterMap.clear();
+                edgesPostAggregationFilterMap.clear();
+                entitiesTransformerMap.clear();
+                edgesTransformerMap.clear();
+                entitiesPostTransformFilterMap.clear();
+                edgesPostTransformFilterMap.clear();
+
+                AddOperationsToChain addOpsHook = new AddOperationsToChain();
+                final java.util.Map<String, List<Operation>> afterOpsMap = new HashMap<>();
+                final List<Operation> hookOpList = new ArrayList<>();
+                OperationView operationView = OperationView.class.cast(op);
+
+                updateView(operationView);
+                operationView.getView().addConfig(ViewValidator.SKIP_VIEW_VALIDATION, "true");
+
+                if (aggregateAfter) {
+                    hookOpList.add(new Aggregate());
+                }
+
+                if (!entitiesPostAggregationFilterMap.isEmpty() || !edgesPostAggregationFilterMap.isEmpty()) {
+                    Filter postAggregationFilter = new Filter.Builder()
+                            .entities(entitiesPostAggregationFilterMap)
+                            .edges(edgesPostAggregationFilterMap)
+                            .build();
+                    hookOpList.add(postAggregationFilter);
+                }
+
+                if (!entitiesTransformerMap.isEmpty() || !edgesTransformerMap.isEmpty()) {
+                    Transform transformFunctions = new Transform.Builder()
+                            .entities(entitiesTransformerMap)
+                            .edges(edgesTransformerMap)
+                            .build();
+                    hookOpList.add(transformFunctions);
+                }
+
+                if (!entitiesPostTransformFilterMap.isEmpty() || !edgesPostTransformFilterMap.isEmpty()) {
+                    Filter postTransformFilter = new Filter.Builder()
+                            .entities(entitiesPostTransformFilterMap)
+                            .edges(edgesPostTransformFilterMap)
+                            .build();
+                    hookOpList.add(postTransformFilter);
+                }
+
+                if (!hookOpList.isEmpty()) {
+                    afterOpsMap.put(op.getClass().getName(), hookOpList);
```

Comment:
We can't do this because if the OperationChain contained:
GetElements
GetElements

We would end up adding the hookOpList after both instances of the GetElements operation.